### PR TITLE
feat(mongodb): backup #3080

### PR DESCRIPTION
**Pull request by cycker**

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=3251) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=3251) before we can accept your contribution.<br/><hr/>**cyckercai** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/blueking-dbm?pullRequest=3251) it.</sub>

File: dbm-ui/backend/flow/consts.py [hunk 1]
```diff
@@ -1271,3 +1271,10 @@ class MongoShardedClusterBackupType(str, StructuredEnum):
 
     MONGOS = EnumField("mongos", _("mongos"))
     SHARD = EnumField("shard", _("shard"))
+
+
+class MongoBackupEnum(str, StructuredEnum):
```
Comment (iSecloud) [inline dbm-ui/backend/flow/consts.py]: 有一个相同的枚举类了，是MongoDBBackupFileTagEnum，保留一个哈

File: dbm-ui/backend/flow/engine/bamboo/scene/mongodb/mongodb_backup.py [hunk 2]
```diff
@@ -36,62 +53,77 @@ def __init__(self, root_id: str, data: Optional[Dict]):
         @param data : 单据传递过来的参数列表，是dict格式
         """
 
-        self.root_id = root_id
-        self.payload = data
+        super().__init__(root_id, data)
+        self.check_payload()
+
+    def check_payload(self):
+        print("payload", self.payload)
```
Comment (iSecloud) [inline dbm-ui/backend/flow/engine/bamboo/scene/mongodb/mongodb_backup.py]: 多余的print语句可以去掉，下面同理。如果要打印日志可以用logger.info之类的

File: dbm-ui/backend/flow/engine/bamboo/scene/mongodb/mongodb_restore.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+# disable W291 trailing whitespace
+def prepare_script_content() -> str:
+    script = """
+# todo add root id and node id
+set -x
+mkdir -p /data/dbbak/recover_mg
+chown -R {} /data/dbbak/recover_mg
+set +x
+echo return code $?"""
```
Comment (iSecloud) [inline dbm-ui/backend/flow/engine/bamboo/scene/mongodb/mongodb_restore.py]: 脚本内容可以考虑维护到一个集中的地方，比如：backend/flow/utils/mongodb/mongodb_script_template.py

File: dbm-ui/backend/flow/engine/bamboo/scene/mongodb/sub_task/remove_ns.py [hunk 4]
```diff
@@ -29,28 +31,41 @@ class RemoveNsSubTask:
...
+    def make_kwargs(cls, payload: Dict, sub_payload: Dict, rs: ReplicaSet, file_path: str) -> dict:
+        print("get_backup_node", sub_payload)
         nodes = rs.get_not_backup_nodes()
         if len(nodes) == 0:
             raise Exception("no backup node. rs:{}".format(rs.set_name))
 
+        ns_filter = sub_payload.get("ns_filter")
+        is_partial = MongoDBNsFilter.is_partial(ns_filter)
         node = nodes[0]
+        dba_user = "dba"
```
Comment (iSecloud) [inline dbm-ui/backend/flow/engine/bamboo/scene/mongodb/sub_task/remove_ns.py:43]: 用户可以用常量维护起来，参考backend/flow/consts.py

如果这个dba用户是mongo创建的，可以考虑带有前缀，比如mongo_dba之类的，避免其他组件用到相同的用户

File: dbm-ui/backend/flow/engine/controller/mongodb.py [hunk 5]
```diff
@@ -40,9 +42,17 @@ def cluster_create(self):
...
         """
-        发起备份任务
+        发起任务
         """
-        flow = MongoBackupFlow(root_id=self.root_id, data=self.ticket_data)
+        # Get Ticket Name. 以后再拆到url那边. 临时用法.
+        ticket_name = self.ticket_data["ticket_type"]
+        if ticket_name == "MONGODB_RESTORE":
+            flow = MongoRestoreFlow(root_id=self.root_id, data=self.ticket_data)
+        elif ticket_name == "MONGODB_FULL_BACKUP" or ticket_name == "MONGODB_PARTIAL_BACKUP":
```
Comment (iSecloud) [inline dbm-ui/backend/flow/engine/controller/mongodb.py:51]: 这个已经有单据类型的枚举类了，可以这样用哈: TicketType.MONGODB_FULL_BACKUP

**Review by iSecloud — APPROVED** [review]

File: dbm-ui/backend/flow/engine/bamboo/scene/mongodb/sub_task/remove_ns.py (hunk 4 above)
Comment (cycker) [reply dbm-ui/backend/flow/engine/bamboo/scene/mongodb/sub_task/remove_ns.py:43]: 后面会改，这次不改

File: dbm-ui/backend/flow/engine/controller/mongodb.py (hunk 5 above)
Comment (cycker) [reply dbm-ui/backend/flow/engine/controller/mongodb.py:51]: 这是临时方案，不想去修改urls.py，下次再改。

**Pull request merged**